Loop/Managers/DeviceDataManager.swift

```diff
@@ -1400,6 +1400,68 @@ extension DeviceDataManager: SupportInfoProvider {
     
 }
 
+//MARK: TherapySettingsViewModelDelegate
+
+extension DeviceDataManager: TherapySettingsViewModelDelegate {
+    
+    func syncBasalRateSchedule(items: [RepeatingScheduleValue<Double>], completion: @escaping (Swift.Result<BasalRateSchedule, Error>) -> Void) {
+        pumpManager?.syncBasalRateSchedule(items: items, completion: completion)
+    }
+    
+    func syncDeliveryLimits(deliveryLimits: DeliveryLimits, completion: @escaping (Swift.Result<DeliveryLimits, Error>) -> Void) {
+        // FIRST we need to check to make sure if we have to cancel temp basal first
+        loopManager.maxTempBasalSavePreflight(unitsPerHour: deliveryLimits.maximumBasalRate?.doubleValue(for: .internationalUnitsPerHour)) { [weak self] error in
+            if let error = error {
+                completion(.failure(SaveTherapySettingsError.cancelTempBasalError(error)))
+            } else if let pumpManager = self?.pumpManager {
+                pumpManager.syncDeliveryLimits(limits: deliveryLimits, completion: completion)
+            } else {
+                completion(.failure(SaveTherapySettingsError.missingPumpManager))
```

Comment:
If there is no pumpmanger, then this should succeed, because settings do not need to be synced to the pump in that case.

Reply:
ha, of course.  Good point 👍 will fix